packages/ui-kit/src/Breadcrumbs/hooks/useModelHook.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState, useLayoutEffect, useRef } from 'react';
+
+interface UseModelProps<T> {
+  items: T[];
+  ellipsisWidth: number;
+}
+
+export const useModel = <T,>({ items, ellipsisWidth }: UseModelProps<T>) => {
```

Comment:
`useModel` is a very generic name. I had to spend some time to realize this hook is specific to Breadcrumbs component. Would be nice to rename it to something more specific to make it more obvious